docs/rfc/rfc-017-abci++-vote-extension-propag.md

```diff
@@ -0,0 +1,438 @@
+in several discussions by the Tendermint Core team.
+
+## Document Structure
+
+The RFC is structured as follows. In the [Background](#background) section,
+subsections [Problem Description](#problem-description) and [Cases to Address](#cases-to-address)
+explain the problem at hand from a high level perspective, i.e., abstracting away from the current
+Tendermint implementation. In contrast, subsection
+[Current Catch-up Mechanisms](#current-catch-up-mechanisms) delves into the details of the current
+Tendermint code.
+
+In the [Discussion](#discussion) section, subsection [Solutions Proposed](#solutions-proposed) is also
+worded abstracting away from implementation details, whilst subsections
+[Feasibility of the Proposed Solutions](#feasibility-of-the-proposed-solutions) and
+[Current Limitations and Possible Implementations](#current-limitations-and-possible-implementations)
+analize the viability of one of the proposed solutions in the context of Tendermint's architecture
+based on reactors. Finally, [Formalization Work](#formalization-work) brifely discusses the work
+still needed demonstrate the correctness of the chosen solution.
+
+## Background
+
+For simplicity in the way sets of validators or votes are expressed in this text, and without loss
+of generality, the remainder of this document assumes that all validators have equal voting power.
+
+### Problem Description
+
+In the version of [ABCI](https://github.com/tendermint/spec/blob/4fb99af/spec/abci/README.md) present up to
+v0.35, for any height *h*, a validator *v* MUST have the decided block *b* and a *commit* for height *h* in
+order to decide at height *h*, and propose at height *h+1* in the rounds of *h+1* where *v* is a proposer.
+This *commit* consists of more than *2n/3* precommit votes voting for *b*, where *n* is the size of the
+validator set at height *h*.
+
+In [ABCI++](https://github.com/tendermint/tendermint/blob/4743a7ad0/spec/abci%2B%2B/README.md),
+the information that a validator *v* MUST have to be able to decide in height *h* is the same:
+the decided block *b* and a *commit* for height *h*. In contrast, *v* MUST have *b* and an
+*extended commit* in order to propose at height *h+1*.
+The *extended commit* is a *commit* where a mandatory vote extension is attached to every
+non-`nil` vote in the *commit*. This mandatory vote extension can be empty, but needs to be signed.
+The signing is carried out with the sending validator's private key (the same as the vote it extends),
+but is independent from the vote, so a vote can be separated from its extension.
+The reason for vote extensions to be mandatory is that, otherwise, an intermediate node could remove
+(and thus censor) the extension of a precommit message.
+
+When a validator takes an active part in consensus at height *h*, it has all the data it needs in its
+`LastCommit` structure to decide on *h* and propose in *h+1*. Things are not so easy in the cases when
```

Comment:
Is the relevant data for deciding on height H in the commit of H. Or if you want to use LastCommit, it is the LastCommit of H + 1 that helps us decide H. I don't find the wording here to differentiate between the two.

Reply:
I see. I think the confusion stems from the fact I mention`LastCommit`, which is a set of votes in Tendermint, and not the `LastCommit` field in the block structure.
Let me try to fix this.